foundry/clientSettings.d.ts

```diff
@@ -83,10 +83,12 @@ declare class ClientSettings {
    * });
    * ```
    */
-  register<M extends string, K extends string>(
+  register<M extends string, K extends string, T>(
     module: M,
     key: K,
-    data: ClientSettings.RegisteredSettings[`${M}.${K}`]
+    data: ClientSettings.Values[`${M}.${K}`] extends boolean | number | string | object
```

Comment:
wouldn't it make more sense to check for
```ts
`${M}.${K}` extends keyof ClientSettings.Values
```
?

Reply:
I believe that would result in that case always being chosen. We'd have to remove `[key: string]: unknown` from the Values interface.

Reply:
I don't think so, since any string is a potential `keyof ClientSettings.Values` (it has `[key: string]: unknown` as a fallback).

Reply:
ah, I understand... maybe we should actually remove `[key: string]: unknown` from `ClientSettings.Values` so that we can actually check it here and explicitly intersect it with `Record<string, unknown>` for `get` and `set`? Or just have additional overloads of those methods?

The problem I see with checking it like you do here is that we need to cover all possible types for data and I think it would be possible for users to add custom types that might not be handled here? What about `symbol`? What about `null`?

Reply:
I'm not sure you can have a setting typed as `null`. I added all of the primitive types here as well as object, which should cover all cases.